packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1762,13 +1853,29 @@ export class SnapController extends BaseController<
     newVersionRange: string = DEFAULT_REQUESTED_SNAP_VERSION,
   ): Promise<TruncatedSnap | null> {
     const snap = this.getExpect(snapId);
+    let newSnap;
 
     if (!isValidSemVerRange(newVersionRange)) {
       throw new Error(
         `Received invalid snap version range: "${newVersionRange}".`,
       );
     }
-    const newSnap = await this.#fetchSnap(snapId, location);
+
+    const permissionsApproval = this.#createApproval({
+      origin,
+      snapId,
+      type: SNAP_APPROVAL_UPDATE,
+    });
+
+    try {
+      newSnap = await this.#fetchSnap(snapId, location);
+    } catch (error) {
+      this.#updateApproval(permissionsApproval.id, {
```

Comment:
This is not enough, we need to also catch potential throws from validation later, e.g. if `#assertIsInstallAllowed`